### PR TITLE
Handle unready relations with status.on_error

### PR DESCRIPTION
**Pull request by addyess**
Let's catch reconcile errors as they happen rather than let than the charm go into error

These changes aren't required to port back to release_1.29

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -112,6 +114,7 @@ def _check_tokens_integration(self, event) -> bool:
             status.add(WaitingStatus(evaluation))
         return False
 
+    @status.on_error(ops.WaitingStatus("Waiting on CNI"))
```
Comment (mateoflorido) [inline src/charm.py:117]: Are any of these methods raising exceptions? I tried to look up each one, but I think all of them are quite safe, unless you don't have the proper CNI related, right? Is that the case we are trying to catch here?

Comment (addyess) [reply src/charm.py:117]: if the CNI is related, but no data is yet available -- they could fail with a KeyError

that's what was going on in the unit tests

**Review by mateoflorido — APPROVED** [review]
Thanks, LGTM!

**Review by kwmonroe — COMMENTED** [review]
I generally approve of this, but since we're reworking for ops.BlockedStatus, should we not also address the [Maint|Waiting] status imports too?  feels like those should all be from ops vs the mix of reconciler / ops.model we have today.

**Comment by addyess** [comment]
> I generally approve of this, but since we're reworking for ops.BlockedStatus, should we not also address the [Maint|Waiting] status imports too? feels like those should all be from ops vs the mix of reconciler / ops.model we have today.

I think i got that too...

**Pull request merged**